src/programs/azimuthal_average/azimuthal_average.cpp

```diff
@@ -0,0 +1,1534 @@
+		else buffer_proj.ApplyCTF(current_ctf);
+
+		buffer_proj.BackwardFFT();
+
+		if (image_counter == 0)
+		{
+			buffer_proj.QuickAndDirtyWriteSlice("my_proj_times_ctf_one.mrc", 1);
+		}
+
+		// subtract mean value at edges to set zero-mean of noise
+		// it would better to calculate noise average from vertical stripes from left and right edges of images
+		average = buffer_proj.ReturnAverageOfRealValues(buffer_proj.physical_address_of_box_center_x - 10.0 / 2.0, true);
+		buffer_proj.AddConstant(-average);
+		//buffer_proj.AddConstant(-buffer_proj.ReturnAverageOfRealValuesOnEdges());
+		//buffer_stack[image_counter].AddConstant(-buffer_stack[image_counter].ReturnAverageOfRealValuesOnEdges());
+
+		// least squares scale factor = A*B / A*A (projection of B onto A)
+		// here A = projection and B = micrograph
+		for (int j = 0; j < buffer_proj.logical_y_dimension; j++)
+		{
+			for (int i = 0; i < buffer_proj.logical_x_dimension; i++)
+			{
+
+				sum_of_pixelwise_product += buffer_proj.real_values[pixel_counter] * buffer_stack[image_counter].real_values[pixel_counter];
+
+				//sum_of_squares += buffer_stack[image_counter].real_values[pixel_counter] * buffer_stack[image_counter].real_values[pixel_counter];
+				sum_of_squares += buffer_proj.real_values[pixel_counter] * buffer_proj.real_values[pixel_counter];
+
+				pixel_counter++;
+			}
+
+			pixel_counter += buffer_proj.padding_jump_value;
+		}
+		
+		scale_factor = sum_of_pixelwise_product / sum_of_squares;
+
+		sum_of_scale_factors += scale_factor;
+		sum_of_scale_factors_squares += powf(scale_factor, 2);
+
+		// multiply by scale factor pixelwise
+		buffer_proj.MultiplyByConstant(scale_factor);
+		//buffer_stack[image_counter].MultiplyByConstant(1./scale_factor);
+
+		// subtract projection from image
+		buffer_stack[image_counter].SubtractImage(&buffer_proj);
+
+		// print out sf*X_i - Proj*CTF_i
+		buffer_stack[image_counter].QuickAndDirtyWriteSlice("my_subtracted_frames.mrc", image_counter + 1);
+
+		// multiply image stack by per image scale factor
+		//input_stack[image_counter].MultiplyByConstant(1./scale_factor);
+	}
+
+	// print average scale factor between images and projection
+	sum_of_scale_factors /= number_of_images;
+	sum_of_scale_factors_squares /= number_of_images;
+	scale_factor_variance = fabsf(float( sum_of_scale_factors_squares - powf(sum_of_scale_factors, 2) ));
+	wxPrintf("mean scale factor = %f, variance scale factor = %f\n", sum_of_scale_factors, scale_factor_variance);
+
+	/*
+	// multiply image stack by average scale factor
+	for (image_counter = 0; image_counter < number_of_images; image_counter++)
+	{
+		input_stack[image_counter].MultiplyByConstant(1./sum_of_scale_factors);
+	}
+	*/
+
+	// multiply reference by average scale factor
+	current_volume->MultiplyByConstant(sum_of_scale_factors);
+
+	// clean-up
+	buffer_proj.Deallocate();
+	delete[] buffer_stack;
+}
+
+void normalize_image(Image* input_image, float pixel_size, float mask_falloff)
```

Comment:
Overall, nice code reuse compared to some of the other sections, but I wonder what this does that wouldn't be accomplished with

Image::Normalise ?

I think the only way the variance is zero is if the image is constant, which is checked for in that method